posthog-web/src/posthog-web.ts

```diff
@@ -19,7 +19,7 @@ export class PostHog extends PostHogCore {
 
     // posthog-js stores options in one object on
     this._storageKey = options?.persistence_name ? `ph_${options.persistence_name}` : `ph_${apiKey}_posthog`
-    this._storage = getStorage(options?.persistence || 'localStorage')
+    this._storage = getStorage(options?.persistence || 'localStorage', window)
```

Comment:
This still needs to be gated because it is still throwing a ReferenceError – you can't just reference `window` in the node.js global scope. 

```
this._storage = getStorage(options?.persistence || 'localStorage', typeof window !== 'undefined' ? window : undefined)
```